zokrates_core/src/static_analysis/constant_inliner.rs

```diff
@@ -176,7 +162,7 @@ impl<'ast, T: Field> Folder<'ast, T> for ConstantInliner<'ast, T> {
     ) -> FieldElementExpression<'ast, T> {
         match e {
             FieldElementExpression::Identifier(ref id) => match self.get_constant(id) {
-                Some(c) => self.fold_expression(c).try_into().unwrap(),
+                Some(c) => c.try_into().unwrap(),
```

Comment:
I suspect we can remove expression folding (`fold_expression`) on other variants like `fold_boolean_expression`, `fold_uint_expression`, etc?

Reply:
like here https://github.com/Zokrates/ZoKrates/blob/759f4dd5534b779c0d73172a94a91ce63ccaac48/zokrates_core/src/static_analysis/constant_inliner.rs#L178